src/reflect/scala/reflect/internal/Trees.scala

```diff
@@ -170,6 +170,27 @@ trait Trees extends api.Trees {
       productIterator foreach subtrees
       if (builder eq null) Nil else builder.result()
     }
+    final def foreachChild(f: Tree => Unit): Unit = {
+      def subtrees(x: Any): Unit = x match {
+        case EmptyTree =>
+        case t: Tree =>
+          f(t)
+        case xs: List[_] =>
+          xs foreach subtrees
+        case _ =>
+      }
+      val N = productArity
+      var i = 0
+      while (i < N) {
+        subtrees(productElement(i))
+        i += 1
+      }
+    }
+
+    /** Equivalent to `this.children.headOption.getOrElse(EmptyTree)`, but more efficient. */
+    final def onlyChild: Tree = {
+      onlyChildAccumulator.using(accum => { foreachChild(accum); accum.result()})
```

Comment:
Would it be worth the effort to break out after seeing the second child?

Reply:
I've added a commit that lets the callback to `foreachChild` stop traversal.